server/pbench/bin/index-pbench

```diff
@@ -514,13 +514,26 @@ _known_csv_handlers = {
             'colpat': re.compile(r'(?P<id>.+)-(?P<subfield>read|write)')
         }
     },
-    'proc-interrupts': {
+    'mpstat': {
         '@prospectus': {
-            # For proc-interrupts .csv files, we want to individually index
-            # each column entry as its own JSON document
+            # For the mpstat .csv files, we want to individually index each
+            # column entry as its own JSON document.
             'handling': 'individual'
         },
+        '@key_mappings': {
+            # The mpstat tool produces csv files with names based on cpu cores.
+            # The amount of cpu cores can be different on each computer, so
+            # map a regular expression to a handler key for each file type.
+            (re.compile(r'(?P<id>cpu.+)_cpu(?P<cpuid>.+)\.csv'), 'generic_cpu'),
```

Comment:
It might be _more_ complicated to use a regular expression as the key in this dictionary. Previously the `get_csv_files()` method used the file name as the key in the dict (`handler_rec = handler[fname]`). If the key can be a regular expression we have to iterate over all keys and attempt a match on the file name, when/if no matches are found then attempt to use the file name as key in the else condition. This is why I added an identifier at the top level.

There is no difference between the id number and `cpu`id. I thought it might come in handy to have the ID NN and the cpuNN for indexing but never used the second one.

Reply:
Can we add something to the `@prospectus` entry to tell direct the code how to work with the handler entries?

It seems like it is worth the effort in the long run, since we'll need this for other tools going forward.

Reply:
I implemented a `handling` key and a `method` key in the prospectus as we discussed.